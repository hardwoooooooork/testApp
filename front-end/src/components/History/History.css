/* History.css */
#History{
    width: 600px;
}