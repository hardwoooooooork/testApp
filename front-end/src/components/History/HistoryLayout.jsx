import React from 'react'
import './History.css'
import ReactTable from 'react-table'
import 'react-table/react-table.css'

export default function HistoryLayout({ historyData, onChangeSearchHandler }) {
    const data = historyData;

    const columns = [{
        Header: 'Keyword',
        width: '300px',
        accessor: 'query' ,
        Cell: props =>
            <p onClick={e => {
                let result = {
                    target : {
                        value: props.value
                    }
                }
                onChangeSearchHandler(result);
            } }>
                {props.value}
            </p>
    }, {
        Header: 'Date',
        width: '600px',
        accessor: 'createdTime',
        Cell: props => 
            <span className='number'>
                {(props.value).replace('T', " ") }
            </span> // Custom cell components!

    },]

     
    return (
        <div id="History">
            <ReactTable
                minRows={5}
                defaultPageSize={5}
                data={data}
                columns={columns}
            />
        </div>
    )
}
