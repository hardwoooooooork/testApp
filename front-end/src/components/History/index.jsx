import React, { Component } from 'react'
import HistoryLayout from './HistoryLayout';

import * as user from '../../services/login'

export default class History extends Component {
    state={ 
        historyData : [],
    }

    componentDidMount(){
        this.getUserHistory().then(historyData => this.setState({historyData}));
    }

    getUserHistory = async ()=>{
        try {
            const res = await user.getUserInfoKeyword();            
            return res.data;
        } catch (error) {
            return undefined;
        }
    }

    render() {
        return (
            <HistoryLayout
                {...this.props}
                {...this.state}
            />
        )
    }
}
