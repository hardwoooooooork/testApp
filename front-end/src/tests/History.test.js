import React from 'react';
import { render } from '@testing-library/react';
import History from '../components/History';

describe('<History />', () => {
    it('matches snapshot', () => {
        const utils = render(<History />);
        expect(utils.container).toMatchSnapshot();
    });
});