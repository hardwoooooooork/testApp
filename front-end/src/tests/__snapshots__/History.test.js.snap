// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> matches snapshot 1`] = `
<div>
  <div
    id="History"
  >
    <div
      class="ReactTable"
    >
      <div
        class="rt-table"
        role="grid"
      >
        <div
          class="rt-thead -header"
          style="min-width: 0300px600pxpx;"
        >
          <div
            class="rt-tr"
            role="row"
          >
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              role="columnheader"
              style="flex: 0 auto;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                Keyword
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              role="columnheader"
              style="flex: 0 auto;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                Date
              </div>
              <div
                class="rt-resizer"
              />
            </div>
          </div>
        </div>
        <div
          class="rt-tbody"
          style="min-width: 0300px600pxpx;"
        >
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -padRow -odd"
              role="row"
            >
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
            </div>
          </div>
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -padRow -even"
              role="row"
            >
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
            </div>
          </div>
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -padRow -odd"
              role="row"
            >
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
            </div>
          </div>
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -padRow -even"
              role="row"
            >
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
            </div>
          </div>
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -padRow -odd"
              role="row"
            >
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
              <div
                class="rt-td"
                role="gridcell"
                style="flex: 0 auto;"
              >
                <span>
                   
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pagination-bottom"
      >
        <div
          class="Table__pagination"
        >
          <div
            class="Table__prevPageWrapper"
          >
            <button
              class="Table__pageButton"
              disabled=""
            >
              Previous
            </button>
          </div>
          <div
            class="Table__visiblePagesWrapper"
          />
          <div
            class="Table__nextPageWrapper"
          >
            <button
              class="Table__pageButton"
            >
              Next
            </button>
          </div>
        </div>
      </div>
      <div
        class="rt-noData"
      >
        No rows found
      </div>
      <div
        class="-loading"
      >
        <div
          class="-loading-inner"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;
